lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -934,7 +934,7 @@ export class UserAgentApplication {
             const navigateWindow: Window = popupWindow ? popupWindow : window;
             const logMessage: string = popupWindow ? "Navigated Popup window to:" + urlNavigate : "Navigate to:" + urlNavigate;
             this.logger.infoPii(logMessage);
-            navigateWindow.location.replace(urlNavigate);
+            navigateWindow.location.assign(urlNavigate);
```

Comment:
Did you consider using `window.history.pushState`? https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

Reply:
history.pushState only works on same domain since it does not reload the page, which wouldn't work here. We could probably do this instead of window.location.href in the handleAuthenticationResponse function however.

Reply:
Ah right of course, forgot this would be a new domain.